sound/soc/sof/topology.c

```diff
@@ -1256,15 +1256,43 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 
 		switch (w->id) {
 		case snd_soc_dapm_dai_out:
-			for_each_rtd_cpu_dai(rtd, i, cpu_dai)
-				cpu_dai->capture_widget = w;
+			for_each_rtd_cpu_dai(rtd, i, cpu_dai) {
+				/*
+				 * The order of DAI widgets defined in topology
+				 * doesnt matter as all the CPU DAI's will be
+				 * enabled/disabled simultaneously
+				 */
+				if (!cpu_dai->capture_widget)
+					break;
+			}
+			if (cpu_dai->capture_widget) {
+				dev_err(scomp->dev, "error: can't find BE for DAI %s\n",
+					w->name);
+
+				return -EINVAL;
+			}
+			cpu_dai->capture_widget = w;
```

Comment:
@bardliao Isnt't the commit message still wrong: " So, modify the loop to set the
playback/capture widget for the first BE CPU DAI that does have a
connection established previously."

What code does seems to be the opposite. It finds first "cpu_dai" that does NOT have a widget set, and assigns the widget to it. The commit says it finds the first "cpu_dai" that already has a widget, and assings the widget stil. This seems wrong.

Reply:
Thanks @bardliao , this is now fixed